src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1679,6 +1667,22 @@ public T next() {
 
     abstract class AbstractIterator<T> implements Iterator<T> {
 
+        public boolean equals(Object o) {
```

Comment:
yes, you are right, interfaces in Java 8 cannot have default methods which override methods of Object. I intentionally left it unoverridden. However, let's do it your way. But we need to check `o instanceof Iterator` IMO. If someone needs to compare an iterable, one could call `iterator.equals(iterable.iterator())`.